examples/src/main/java/io/dapr/examples/invoke/grpc/HelloWorldClient.java

```diff
@@ -45,22 +44,11 @@ public class HelloWorldClient {
    * @param args Array of messages to be sent.
    */
   public static void main(String[] args) throws Exception {
-
     String user = "World";
-    String target = "localhost:" + System.getenv("DAPR_GRPC_PORT");
-
-    ManagedChannel channel = Grpc.newChannelBuilder(target, InsecureChannelCredentials.create())
-        .build();
-
-    try {
-      HelloWorldGrpc.HelloWorldBlockingStub blockingStub = HelloWorldGrpc.newBlockingStub(channel);
-
-      Metadata headers = new Metadata();
-      headers.put(Metadata.Key.of("dapr-app-id", Metadata.ASCII_STRING_MARSHALLER),
-          "hellogrpc");
-
-      // MetadataUtils.attachHeaders is deprecated.
-      blockingStub = blockingStub.withInterceptors(MetadataUtils.newAttachHeadersInterceptor(headers));
+    try (DaprClient client = new DaprClientBuilder().build()) {
+      HelloWorldGrpc.HelloWorldBlockingStub blockingStub = HelloWorldGrpc.newBlockingStub(client.getGrpcChannel());
+      // Adds Dapr interceptors to populate gRPC metadata automatically.
+      blockingStub = DaprClientGrpcInterceptors.intercept("hellogrpc", blockingStub);
```

Comment:
This is the new way to invoke gRPC, now using the gRPC proxy and with the helper methods: `client.getGrpcChannel()` and `DaprClientGrpcInterceptors.intercept(appId, stub);`

Reply:
I was wondering if I should offer a new API that takes a callback function instead.

Reply:
See the new design. I think it is cleaner.